test/sql/catalog/test_set_search_path.test

```diff
@@ -0,0 +1,141 @@
+# name: test/sql/catalog/test_set_search_path.test
+# description: SET schema and SET search_path
+# group: [catalog]
+
+# create a couple of schemas with a table each
+statement ok
+CREATE TABLE main_table(i INTEGER);
+
+statement ok
+CREATE SCHEMA test;
+
+statement ok
+CREATE TABLE test.test_table(i INTEGER);
+
+statement ok
+CREATE SCHEMA test2;
+
+statement ok
+CREATE SCHEMA test3;
+
+statement ok
+CREATE TABLE test2.bye(i INTEGER);
+CREATE TABLE test2.test2_table(i INTEGER);
+
+# Reading the config - It should be empty for now.
+statement error
+SELECT CURRENT_SETTING('search_path');
+
+statement error
+SELECT CURRENT_SETTING('schema');
+
+# Setting the default value.
+statement ok
+SET SEARCH_PATH = 'test';
```

Comment:
Could we add tests for:
* A single quoted value (`"test"`)
* A single entry with a quote in the middle of it (`hello"world`)
* Many entries (e.g. 5 entries)
* Various empty entries (e.g. the following strings: `''`  `',,'` `'""'` `'"",""'` `'hello,""'`)
* Duplicate entries in the search path (`main,main`) (should this be allowed?)
* A test with multiple connections that verifies setting the SEARCH_PATH is local to each connection (see e.g. `test/sql/transactions/test_index_pending_update.test` for a test that uses multiple connections)

Reply:
> Could we add tests for:
> 
> * A single quoted value (`"test"`)
> * A single entry with a quote in the middle of it (`hello"world`)
> * Various empty entries (e.g. the following strings: `''`  `',,'` `'""'` `'"",""'` `'hello,""'`)
These are (already/newly) added to the C++ unit test. It's hard to test these from SQL since it's not clear if the code path is correctly exercised or if it is rejected due to the missing schema name.

> * Many entries (e.g. 5 entries)
> * Duplicate entries in the search path (`main,main`) (should this be allowed?)
> * A test with multiple connections that verifies setting the SEARCH_PATH is local to each connection (see e.g. 
Done. I thinks having duplicate is fine. The semantics is clear enough.